### PR TITLE
Actually adjust copy button logic to new API

### PR DESCRIPTION
**Pull request by D3SOX**
The previous code (introduced in #6698) contained obsolete logic from the deprecated clipboard API, which was removed here.

The Promise returned by `navigator.clipboard.writeText` is now correctly handled as described per https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/writeText#return_value

I've also removed the check if it's running as test as [HTMLUnit 3.1.0](https://htmlunit.sourceforge.io/changes-report.html#a3.1.0) added support for `isSecureContext`
### Testing done

I have overwritten the function with the console to always return a rejecting Promise. Therefore, the new logic is tested
image

### Proposed changelog entries

I think this does not need to be mentioned in the changelog

### Alternatives I have considered

We could implement the deprecated clipboard API again when we are not in a secure context

## Additional testing done

* [x] @MarkEWaite confirm that the copy button behaves as expected with a build of this pull request

### CODE REVIEW

File: core/src/main/resources/lib/layout/copyButton/copyButton.js
Comment (D3SOX) [inline core/src/main/resources/lib/layout/copyButton/copyButton.js:9]:
```suggestion
      if (isSecureContext) {
```

I just found out that [HTMLUnit 3.1.0](https://htmlunit.sourceforge.io/changes-report.html#a3.1.0) added support for `isSecureContext`. Which version is currently used for testing? Maybe this can be removed like this

Comment (MarkEWaite) [reply core/src/main/resources/lib/layout/copyButton/copyButton.js:9]: We testing with HTMLUnit 3.5.0 per https://github.com/jenkinsci/jenkins-test-harness-htmlunit/releases/tag/154.v973ee207a_b_95

**Review by yaroslavafenkin — COMMENTED** [review]
Looks OK from security POV.

**Comment by MarkEWaite** [comment]
Thanks very much @D3SOX .  I've submitted a verification test with the plugin bill of materials to confirm that this improvement does not surprise us in the bill of materials release.  Pull request is:

* https://github.com/jenkinsci/bom/pull/2562

**Review by MarkEWaite — APPROVED** [review]
I tested the copy button on an SSL enabled Jenkins behind an NGINX reverse proxy and it worked great.  Approved.  Thanks very much.

Tests also ran as expected on the Jenkins plugin bill of materials.  No test failures reported.

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

File: core/src/main/resources/lib/layout/copyButton/copyButton.js
Comment (jglick) [inline core/src/main/resources/lib/layout/copyButton/copyButton.js:7]: introduced in https://github.com/jenkinsci/jenkins/pull/7784 FTR

File: core/src/main/resources/lib/layout/copyButton/copyButton.js
Comment (jtnord) [inline core/src/main/resources/lib/layout/copyButton/copyButton.js:9]: whilst mainstream browsers support this (according to canIUse) I believe unless you set a handler in `HtmlUnit` that this will be undefined when runnign tests with webclient.

the [webclient created in j-t-h](https://github.com/jenkinsci/jenkins-test-harness/blob/master/src/main/java/org/jvnet/hudson/test/JenkinsRule.java#L2329-L2333) is not configured with a handler.

**Pull request merged**
